packages/notionable/src/models/points-store.ts

```diff
@@ -0,0 +1,47 @@
+import { flow, Instance, types } from 'mobx-state-tree';
+
+const ArbPoint = types.model({
+  token: types.string,
+  points: types.number,
+  season_one: types.number,
+  season_two: types.number,
+  season_three: types.number,
+});
+
+export const PointsStoreModel = types
+  .model({
+    arbPoints: types.array(ArbPoint),
+    totalPoints: types.optional(types.number, 0),
+  })
+  .actions((self) => {
+    const fetchPoints = flow(function* (selectedAddress) {
+      try {
+        const response = yield fetch(
+          `https://points.notional.finance/arb_account_points/${selectedAddress.toLowerCase()}`
+        );
+        const arbPoints = yield response.json();
+        self.arbPoints = arbPoints.map((point: any) => ArbPoint.create(point));
+        self.totalPoints = self.arbPoints.reduce(
+          (t, { points }) => t + points,
+          0
+        );
+      } catch (error) {
+        console.warn('Error fetching arb points', error);
+      }
+      return self.arbPoints;
+    });
+
+    const initialize = (selectedAddress: string) => {
+      if (selectedAddress) {
+        fetchPoints(selectedAddress);
+      }
```

Comment:
In the else condition you will need to set the values back to the initial values.